doc/upload.md

```diff
@@ -0,0 +1,179 @@
+
+Take a look at at `manifests/example/upload-pvc.yaml` for an example.
+```yaml
+apiVersion: v1
+kind: PersistentVolumeClaim
+metadata:
+  name: upload-test
+  labels:
+    app: containerized-data-importer
+  annotations:
+    cdi.kubevirt.io/storage.upload.target: ""
+spec:
+  accessModes:
+  - ReadWriteOnce
+  resources:
+    requests:
+      storage: 1Gi
+
+```
+```bash
+kubectl apply -f manifests/example/upload-pvc.yaml
+```
+
+## Request an Upload Token
+Before sending data to the Upload Proxy, and Upload Token must be requested.  The CDI API Server validatees that the user has permissions to `post` to `uploadtokenrequest` resources.
+
+Take a look at at `manifests/example/upload-token.yaml` for an example.
+```yaml
+apiVersion: upload.cdi.kubevirt.io/v1alpha1
+kind: UploadTokenRequest
+metadata:
+  name: upload-test
+  namespace: default
+spec:
+  pvcName: upload-test
+
+```
+```bash
+kubectl apply -f manifests/example/upload-token.yaml -o yaml
+apiVersion: upload.cdi.kubevirt.io/v1alpha1
+kind: UploadTokenRequest
+metadata:
+  annotations:
+    kubectl.kubernetes.io/last-applied-configuration: |
+      {"apiVersion":"upload.cdi.kubevirt.io/v1alpha1","kind":"UploadTokenRequest","metadata":{"annotations":{},"name":"upload-test","namespace":"default"},"spec":{"pvcName":"upload-test"}}
+  creationTimestamp: null
+  name: upload-test
+  namespace: default
+spec:
+  pvcName: upload-test
+status:
+  token: eyJhbGciOiJQUzUxMiIsImtpZCI6IiJ9.eyJwdmNOYW1lIjoidXBsb2FkLXRlc3QiLCJuYW1lc3BhY2UiOiJkZWZhdWx0IiwiY3JlYXRpb25UaW1lc3RhbXAiOiIyMDE4LTA5LTIxVDE4OjEyOjE5LjQwODI1MDQ4NFoifQ.JWk1VyvzSse3eFiBROKgGoLnOPCiYW9JdDWKXFROEL6XY0O5lFb1R0rwdfWwC3BBOtEA9mC9x3ZGYPnYWO-5G_r1fWKHjF-zifrCX_3Dhp3vfSq6Zfpu-vV0Qn0A3YkSCCmiC_nONAhVjEDuQsRFIKwYcxBoEOpye92ggH2u5FxQE7FwxxH6-RHun9tc_lIFX-ZFKnq7n5tWbjsTmAZI_4rDNgYkVFhFtENU6e-5_Ncokxs3YVzkbSrXweZpRmmaYQOmZhjXSLjKED_2FVq7tYeVueEEhKC_zJ-AEivstALPwPjiwyWXJyfE3dCmbA1sBKuNUrAaDlBvSAp1uPV9eQ
+  ```
+  Save the `token` field of the response status.  It will be used to authorize our CDI Upload request. Tokens are good for 5 minutes.
+
+You can capture the token in an environment variable by doing this:
+```bash
+TOKEN=$(kubectl apply -f manifests/example/upload-token.yaml -o="jsonpath={.status.token}")
+``` 
+
+## Upload an Image
+We will be using [curl](https://github.com/curl/curl) to upload `tests/images/cirros-qcow2.img` to the PVC.
+
+Assuming that the environment variable `TOKEN` contains a valid UploadToken, execute the following to upload the image:
+
+### Minikube
+```bash
+curl -v --insecure -H "Authorization: Bearer $TOKEN" --data-binary @tests/images/cirros-qcow2.img https://$(minikube ip):31001/v1alpha1/upload
+```
+
+### Minishift
+
+```bash
+curl -v --insecure --resolve cdi-uploadproxy:443:$(minishift ip) -H "Authorization: Bearer $TOKEN" --data-binary @tests/images/cirros-qcow2.img https://cdi-uploadproxy/v1alpha1/upload
+```
```

Comment:
Should we update this example with the nip.io  hostname that works with minishift 1.24?

Reply:
Yes